### DISCUSSION TITLE
Is there any simple way to migrate all our existing NuGet packages to GitHub Package Registry?

### DISCUSSION THREAD

**Discussion by ws-markb** [community/community#26754] [Packages]
<p>I’m looking to move all of our company’s NuGet packages to GPR; we’re migrating all our git repos into GitHub, so it makes sense to me to host all our artifacts in GitHub too.</p>
<p>However, I can’t currently see a way I can get all of the historical versions from our current, self-hosted (but otherwise standard) NuGet feed into the GitHub Package Registry. </p>
<p>What I’d like to do ideally, on a repo-by-repo basis, is:</p>
<ol>
<li>Migrate the Git repository into GitHub</li>
<li>Point something at our current NuGet feed (or run a script) which will get all the versions of the package corresponding to that repository, and then push them to GPR</li>
</ol>
<p>After some research, I found that there’s a <a href="https://docs.microsoft.com/en-us/azure/devops/artifacts/tutorials/migrate-packages?view=azure-devops" rel="nofollow noopener">Powershell module</a> which would help if we wanted to migrate to Azure Artifacts, but before I try hacking about with that code or writing something from scratch myself, I thought it was worth asking here!</p>
<p>Is there anything like this available, or has anyone else scripted this yet?</p>

**Comment by jcansdale** [comment, marked as answer]
<blockquote>
<p>Is there anything like this available, or has anyone else scripted this yet?</p>
</blockquote>
<p>The trick will be to update the <code>&lt;repository url="..." /&gt;</code> element in the .nuspec file in your .nupkg. This is mandatory when publishing to GitHub Packages and tells the registry which repository to associate with your package.</p>
<p>I have a .NET Core tool that supports publishing NuGet packages to the GitHub Package registry (and a few other things). You can find it here: <a href="https://github.com/jcansdale/gpr" rel="nofollow noopener">https://github.com/jcansdale/gpr</a></p>
<p>There’s an issue open about automatically adding the repository url on push:</p>
<ul>
<li>Feature: Set default repository and automatically add repository url and type on push<br>
<a href="https://github.com/jcansdale/gpr/issues/48" rel="nofollow noopener">https://github.com/jcansdale/gpr/issues/48</a>
</li>
</ul>
<p>Feel free to comment or send a PR my way! :smile:</p>

**Comment by JamesWebDev** [comment]
<pre><code class="lang-auto">dotnet tool install gpr -g
gpr push MyFakePackage.1.0.0.50.nupkg --repository https://github.com/MyFakeOrg/my-repo-name
</code></pre>
<p>I tried it and can confirm it works. It rewrites the nuspec file and adds the node that github requires</p>

**Reply by joshjohanning** [reply]
Nice! I expanded this a bit to create a script that will push and map `nupkg` files to GitHub packages: https://josh-ops.com/posts/github-packages-migrate-nuget-packages-to-github-packages/
